dist/js/fuelux.js

```diff
@@ -2956,6 +2959,271 @@
+			},
+
+			destroy: function() {
+				this.$element.remove();
+				// remove any external bindings
+				$( document ).off( 'click.fu.superpicker.externalClick.' + this.clickStamp );
+				// set input value attrbute
+				this.$element.find( 'input' ).each( function() {
+					$( this ).attr( 'value', $( this ).val() );
+				} );
+				// empty elements to return to original markup
+				// [none]
+				// return string of markup
+				return this.$element[ 0 ].outerHTML;
+			},
+
+			disable: function() {
+				this.$element.addClass( 'disabled' );
+				this.$field.attr( 'disabled', 'disabled' );
+				this.hide();
+			},
+
+			enable: function() {
+				this.$element.removeClass( 'disabled' );
+				this.$field.removeAttr( 'disabled' );
+			},
+
+			externalClickListener: function( e, force ) {
+				if ( force === true || this.isExternalClick( e ) ) {
+					this.complete( this.options.externalClickAction );
+				}
+			},
+
+			hide: function() {
+				if ( !this.$element.hasClass( 'showing' ) ) {
+					return;
+				}
+
+				this.$element.removeClass( 'showing' );
+				this.$field.removeAttr( 'disabled' );
+				$( document ).off( 'click.fu.superpicker.externalClick.' + this.clickStamp );
+				this.$element.trigger( 'hidden.fu.superpicker' );
+			},
+
+			isExternalClick: function( e ) {
+				var el = this.$element.get( 0 );
+				var exceptions = this.options.externalClickExceptions || [];
+				var $originEl = $( e.target );
+				var i, l;
+
+				if ( e.target === el || $originEl.parents( '.superpicker:first' ).get( 0 ) === el ) {
+					return false;
+				} else {
+					for ( i = 0, l = exceptions.length; i < l; i++ ) {
+						if ( $originEl.is( exceptions[ i ] ) || $originEl.parents( exceptions[ i ] ).length > 0 ) {
+							return false;
+						}
+
+					}
+				}
+
+				return true;
+			},
+
+			show: function() {
+				var other;
+
+				if ( this.$element.hasClass( 'showing' ) ) {
+					return;
+				}
+
+				other = $( document ).find( '.superpicker.showing' );
+				if ( other.length > 0 ) {
+					if ( other.data( 'fu.superpicker' ) && other.data( 'fu.superpicker' ).options.explicit ) {
+						return;
+					}
+
+					other.superpicker( 'externalClickListener', {}, true );
+				}
+
+				this.previousValue = this.$field.val();
+
+				this.$element.addClass( 'showing' );
+
+				this.$popup.css( 'top', ( this.$field.outerHeight( true ) + 4 ) + 'px' );
+
+				this.$popup.css( 'height', ( this.options.height ) ? this.options.height : DEFAULT_HEIGHT + 'px' );
+				this.$body.css( 'height', ( ( this.options.height ) ? this.options.height : DEFAULT_HEIGHT ) - 73 + 'px' );
+				this.$popup.css( 'width', ( this.options.width ) ? this.options.height : DEFAULT_WIDTH + 'px' );
+
+				this.$element.trigger( 'shown.fu.superpicker', this.actualValue );
+				if ( this.actualValue !== null ) {
+					this.actualValue = null;
+				}
+
+				this.$field.attr( 'disabled', 'disabled' );
+
+				this.clickStamp = new Date().getTime() + ( Math.floor( Math.random() * 100 ) + 1 );
```

Comment:
Why generate this each time on show instead of in the constructor? (And for the matter, I assume the purpose of the `clickStamp` is to isolate your `.off` calls, but maybe you could achieve the same effect by including the function in that call?)